testsuite/objects/__init__.py

```diff
@@ -86,15 +93,21 @@ class Value(ABCValue):
     value: JSONValues
 
 
+@dataclass
+class _AuthJson:
```

Comment:
I don't understand this, a brief note in commit message would help (I am sorry if it is already there), it has no other use than in ValueFrom, so the value/benefit is unclear

Also I suggest in these cases to follow more strictly naming scheme of underlying objects as much as possible/reasonable. So I would keep capital 'A' however I would keep JSON capitalized so 'AuthJSON' is in my opinion best option.

Why 'private' with '_'?

Reply:
I renamed the class and unmarked it as private. 
I was at first opposing creation of new class just to solve the custom asdict function, but after extending functionality of `ABCValue` I did not find a way to make the custom asdict function to work so I resorted to creating this new class to solve the problem. Using `__post_init__` to transform `str` to instance of `AuthJSON` class makes using the class still straightforward.
For example
```py
from from testsuite.objects import ValueFrom

my_value = ValueFrom("somewhere.here")
```
instead of:
```py
from from testsuite.objects import ValueFrom, AuthJSON

my_value = ValueFrom(AuthJSON("somewhere.here"))
```
But in the end both ways will work in current code.

I made it private at first because I did not see the second usage in my example as very real.

Reply:
In my recent push I remade the the logic once more. I removed the new class and used [Init-only variable](https://docs.python.org/3/library/dataclasses.html#init-only-variables) and subsequent transformation in `__post_init__`

Reply:
I reverted to original. Will keep the thought for future.